projects/addon-charts/components/line-chart/line-chart-hint.directive.ts

```diff
@@ -63,14 +63,17 @@ export class TuiLineChartHint implements AfterViewInit {
     }
 
     // _chart is required by TuiLineDaysChartComponent that impersonates this directive
-    public getContext(index: number, _chart: TuiLineChart): readonly TuiPoint[] {
+    public getContext(
+        index: number,
+        _chart: TuiLineChart,
+    ): readonly TuiPoint[] | undefined {
         return this.computeContext(...this.charts.map(({value}) => value))[index];
     }
 
     // _chart is required by TuiLineDaysChartComponent that impersonates this directive
     public raise(index: number, _chart: TuiLineChart): void {
         const current = this.charts.map((chart) => chart.value[index]);
-        const sorted = [...current].sort((a, b) => a[1] - b[1]);
+        const sorted = [...current].sort((a, b) => (a?.[1] || 0) - (b?.[1] || 0));
```

Comment:
chart.value is an array of guaranteed tuples, we should be able to add [0, 0] fallback on the line above when we map, rather than using ?. and fallbacks here.